fpmsyncd/bgpmon.py

```diff
@@ -0,0 +1,171 @@
+        # handle IPV4 case
+        for i in range (0, len(self.new_ipv4_n)):
+            neighb = self.new_ipv4_n[i]
+            key = "NEIGH_STATE_TABLE|%s" % neighb
+            if neighb in self.ipv4_n:
+                # only update the entry if sate changed
+                if self.ipv4_n_state[neighb] != self.new_ipv4_n_state[neighb]:
+                    # state changed. Update state DB for this entry
+                    state = self.new_ipv4_n_state[neighb]
+                    self.db.set(self.db.STATE_DB, key, 'state', state)
+                    self.ipv4_n_state[neighb] = state
+                # remove this neighbor from old list since it is accounted for
+                self.ipv4_n.remove(neighb)
+            else:
+                # New neighbor found case. Add to dictionary and state DB
+                state = self.new_ipv4_n_state[neighb]
+                self.db.set(self.db.STATE_DB, key, 'state', state)
+                self.ipv4_n_state[neighb] = state
+        # Check for stale state entries to be cleaned up
+        while len(self.ipv4_n) > 0:
+            # remove this from the stateDB and the current nighbor state entry
+            neighb = self.ipv4_n.pop(0)
+            del_key = "NEIGH_STATE_TABLE|%s" % neighb
+            self.db.delete(self.db.STATE_DB, del_key)
+            del self.ipv4_n_state[neighb]
+        # Save the new List
+        self.ipv4_n = self.new_ipv4_n[:]
+
+        # handle IPV6 case
+        for i in range (0, len(self.new_ipv6_n)):
+            neighb = self.new_ipv6_n[i]
+            key = "NEIGH_STATE_TABLE|%s" % neighb
+            if neighb in self.ipv6_n:
+                # only update the entry if sate changed
+                if self.ipv6_n_state[neighb] != self.new_ipv6_n_state[neighb]:
+                    # state changed. Update state DB for this entry
+                    state = self.new_ipv6_n_state[neighb]
+                    self.db.set(self.db.STATE_DB, key, 'state', state)
+                    self.ipv6_n_state[neighb] = state
+                # remove this neighbor from old list since it is accounted for
+                self.ipv6_n.remove(neighb)
+            else:
+                # New neighbor found case. Add to dictionary and state DB
+                state = self.new_ipv6_n_state[neighb]
+                self.db.set(self.db.STATE_DB, key, 'state', state)
+                self.ipv6_n_state[neighb] = state
+        # Check for stale state entries to be cleaned up
+        while len(self.ipv6_n) > 0:
+            # remove this from the stateDB and the current nighbor state entry
+            neighb = self.ipv6_n.pop(0)
+            del_key = "NEIGH_STATE_TABLE|%s" % neighb
+            self.db.delete(self.db.STATE_DB, del_key)
+            del self.ipv6_n_state[neighb]
+        # Save the new List
+        self.ipv6_n = self.new_ipv6_n[:]
+
+def main():
+
+    print "bgpmon service started"
+
+    try:
+        bgp_state_get = BgpStateGet()
+    except Exception, e:
```

Comment:
```suggestion
    except Exception as e:
```

Reply:
@qiluo-msft Agreed. Will change it per suggested.